webapp/src/services/index.ts

```diff
@@ -11,22 +11,29 @@ const pluginApi = createApi({
     baseQuery: fetchBaseQuery({baseUrl: Utils.getBaseUrls().pluginApiBaseUrl}),
     tagTypes: ['Posts'],
     endpoints: (builder) => ({
-        [Constants.pluginApiServiceConfigs.createTask.apiServiceName]: builder.query<void, CreateTaskPayload>({
+        [Constants.pluginApiServiceConfigs.createTask.apiServiceName]: builder.query<void, APIRequestPayload>({
```

Comment:
Since, this API is for `createTask`, shouldn't payload type be `CreateTaskPayload` rather than `APIRequestPayload`.

Reply:
we have created one type `APIRequestPayload` which is union of different types, if you see type file we have `type APIRequestPayload = CreateTaskPayload | LinkPayload | void;` this is to generalise changes